src/LIBSVM.jl

```diff
@@ -262,6 +254,20 @@ function set_num_threads(nt::Integer)
     ccall((:svm_set_num_threads, libsvm), Cvoid, (Cint,), nt)
 end
 
+function check_dims(X, y, kernel)
+    if kernel == Kernel.Precomputed
+        if size(X, 1) != size(X, 2)
+            throw(ArgumentError("The input matrix must be square"))
```

Comment:
Is `DimensionMismatch` better?

Reply:
I agree. Added a simple test for it too.